packages/explorer/src/common.ts

```diff
@@ -1,22 +1,36 @@
 import { Chain, anvil, garnet, redstone } from "viem/chains";
 
-export const chains: Partial<Record<number, Chain>> = {
+const supportedChains = [anvil, garnet, redstone];
+
+export type SupportedChainIds = (typeof supportedChains)[number]["id"];
+export type SupportedChainNames = "anvil" | "garnet" | "redstone";
+
+export const chainIdName: Record<SupportedChainIds, SupportedChainNames> = {
+  [anvil.id]: "anvil",
+  [garnet.id]: "garnet",
+  [redstone.id]: "redstone",
+} as const;
+
+export const chainNameId: Record<SupportedChainNames, SupportedChainIds> = {
+  anvil: anvil.id,
+  garnet: garnet.id,
+  redstone: redstone.id,
+};
```

Comment:
there's nothing in these types to ensure we maintain parity with the above map

can we compute this or add type constraints to make sure they stay aligned?

Reply:
Initially computed with `reduce` but thought it's too much boilerplate for that. However, your comment made me realize that keeping it aligned is definitely preferred, will change. This is where `_.lodash` is nice but will make without it.